### DISCUSSION TITLE
[v2.0] Frequency in Numerical habits

### DISCUSSION THREAD

**Discussion by kumaraish** [iSoron/uhabits#939] [Feature Requests]
Frequency is limited to "Every day", "Every week" and "Every month" for numerical (measurable) habits. 

Is there a reason why a frequency like "Every X days" or "X times per Week/Month" is not supported for numerical habits?
They are provided for Boolean (Yes/No) habits.

Simple case - "Spent 3 hours learning French, 2 times per Week"

**Comment by Silviopina** [comment]
I would like to see reminders customizable to days of frequency as opposed to days of the week. I have something that needs to happen every 4 days and I cannot customize my notifications reminders to every 4 days but I have to pick arbitrary days of the week ☺️

**Comment by ke352802081770314** [comment]
indeed I see no reason why the same editor for frequency could not be used for both numerical and non-numerical habits. It would be more consistent that way. 
On second thought, I do see two issues. 
One, the goal for numerical habits includes a decimal point while the number in the non-numeric per-week/per month dialogue does not.
Two, we have two numbers interacting with each other, the goal on the main edit screen and the number in the per-period dialogue. The logical thing in my opinion would be that I have a daily  goal in the main edit screen, which governs whether that day "counts" or not, but the "per week" / "per month" would not be days required but total required. However, for the "every N days" option, we are lacking the second number. This may tie in nicely with #961 .
Example: "go for a walk" daily minimum is 10 minutes, and in a week I set myself a goal of 120 minutes. So to make a day count, its number must be 10 or more (because otherwise it isn't recognizable as a walk) but beyond that I can shift the 120 minutes as convenient, so a week with 15, 0, 8 (not counted), 0, 20, 45, 60 has a counted total of 140 minutes and therefore is good.

I haven't figured out yet how the app visualizes goal reached in multi-day numerical goals. I see for per-day goals a number will be gray if below and colored if at least the goal. But for multi-day goals it seems to me any number will be colored, but in-between days will not, so there is no equivalent to the "outline tick mark" within periods of satisfying the goal.

Maybe the streak could be visualized uniformly between non-numerical and numerical habits by a colored line along the botton of the row or something like that.
I guess it was to be expected that a new feature like numerical habits cannot be as mature right from the start as the established functionality, and I am looking forward to seeing it improve further. Sorry for mixing some different thoughts into this reply. Maybe we should have a discussion thread "numerical habits improvements" to bundle thought on the topic. or would you prefer if I open separate threads for each aspect @iSoron ?

**Reply by herrbrixter** [reply]
This is confusing.

**Comment by amrsamii** [comment]
Indeed it will be good to be able to set a frequency for a measurable habit like yes/no habit! I also want to set a different goal for each day. For example, I have a measurable habit (reading) that should be done three times per week. I want to set the goal for two of them to be 10 pages (I will be busy those days), and for the last day (in the weekend), I want to set the goal to be 20 pages.

**Comment by robrys** [comment]
@iSoron Curious for your thoughts as unifying the two frequencies for Yes/No goals and Measurable goes seems in line with the minimalism of the overall app.

**Comment by herrbrixter** [comment]
I'd like this too. For example:

"Eat at least 3 fruits 2 times per week."

2 days in a week, the user should have eaten at least 3 fruits. If he ate only 1 fruit that day, it doesn't count.
